base/twiceprecision.jl

```diff
@@ -107,6 +107,9 @@ end
 function floatrange(a::AbstractFloat, st::AbstractFloat, len::Real, divisor::AbstractFloat)
     T = promote_type(typeof(a), typeof(st), typeof(divisor))
     m = maxintfloat(T)
```

Comment:
Yes, that's a more code-efficient way of doing the check. I'll update this PR along those lines.